src/core/event/index.js

```diff
@@ -18,14 +18,20 @@ export function Events(Base) {
 
       // If 'history', rely on the browser's scroll auto-restoration when going back or forward
       if (source !== 'history') {
```

Comment:
Does it work in history mode?

Reply:
It will work the same way it does on initial page load of a site in `hash` mode.

Reply:
Confirmed working as expected.

FYI: In the highlighted code, `history` is referring to the type of navigation that is causing the URL to change (e.g., clicked link vs browser back/forward functionality) and not Docsify's [routerMode](https://docsify.js.org/#/configuration?id=routermode) configuration option.